pyk/src/pyk/integration_tests/test_kprint.py

```diff
@@ -0,0 +1,30 @@
+from ..kast import KApply, KToken
+from ..ktool import KompileBackend
+from ..prelude import intToken
+from .kprove_test import KProveTest
+
+
+class ImpParseTest(KProveTest):
+    KOMPILE_MAIN_FILE = 'k-files/imp.k'
+    KOMPILE_BACKEND = KompileBackend.HASKELL
+    KOMPILE_OUTPUT_DIR = 'definitions/imp'
+    KOMPILE_EMIT_JSON = True
+
+    KPROVE_USE_DIR = '.imp'
+
+    @staticmethod
+    def _update_symbol_table(symbol_table):
+        pass
+
+    def test_parse_token(self):
+        test_parses = (
+            ('int-token', KToken('3', 'Int'), intToken(3)),
+            ('id-token', KToken('abc', 'Id'), KToken('abc', 'Id')),
+            ('add-aexp', KToken('3 + 4', 'AExp'), KApply('_+_', [intToken(3), intToken(4)])),
+            # TODO: ('k-cell', KToken('<k> . </k>', 'KCell'), KApply('<k>', [KApply(Labels.EMPTY_K)])),
+        )
```

Comment:
While I love that we can use `kast` here this completely blocks the KIT upstream PR. The `case-split` command requires that we are able to parse k variables.

Reply:
Let me know if https://github.com/runtimeverification/k/pull/2787#issuecomment-1210258327 and https://github.com/runtimeverification/k/pull/2787/commits/7c68dfd46d4490d78ab20aa161cec3a411b6ec51 is a reasonable solution to this situation.

Reply:
I mean, yeah, that's fine, but I wanted to get a cleaner version merged here, and then open the relevant issues on the frontend, and then get a hacky version merged.

I'll incorporate a hacky rule-parser version here as well.

Reply:
I didn't have time to make the hacky version. I would prefer to just get this merged for now, with the tests marking what we need.

Reply:
Marking this as resolved to get this merged.